From 761836bfe162aeec61c4cffae0f810083fd3c93a Mon Sep 17 00:00:00 2001
From: qianbi1999 <zhoujian157@huawei.com>
Date: Wed, 26 Nov 2025 10:21:25 +0800
Subject: [PATCH 3/3] Modify compute_score to encourage tool call

In the `compute_score` function within `verl\recipe\retool\retool.py`, adjusted the reward scoring logic to explicitly encourage the model to invoke tools.
---
 recipe/retool/retool.py | 8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/recipe/retool/retool.py b/recipe/retool/retool.py
index 7bcc7045..036bd276 100644
--- a/recipe/retool/retool.py
+++ b/recipe/retool/retool.py
@@ -104,15 +104,15 @@ class CustomRLHFDataset(RLHFDataset):
         return row
 
 
-def compute_score(data_source, solution_str, ground_truth, extra_info):
+def compute_score(data_source, solution_str, ground_truth, extra_info, **kwargs):
     # use \\boxed{...} answer
     result = math_dapo.compute_score(solution_str, ground_truth, strict_box_verify=True)
 
     # encourage model to call tools
     num_turns = extra_info["num_turns"]
-    if result["score"] < 0:
-        tool_call_reward = (num_turns - 2) / 2 * 0.1
-        result["score"] = min(-0.6, result["score"] + tool_call_reward)
+    tool_call_num = (num_turns - 2) // 2
+    if tool_call_num == 0:
+        result["score"] -= 0.4
 
     if result["pred"] is None:
         result["pred"] = ""
-- 
2.50.1.windows.1

